pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -117,25 +128,63 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	}
 
 	// Connect to oVirt:
-	ovirtSecret, err := r.fetchOvirtSecret(instance)
+	ovirtSecretObj, err := r.fetchOvirtSecret(instance)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	ovirtConnection, err := createOvirtConnection(ovirtSecret)
+
+	ovirtSecretDataMap := make(map[string]string)
+	err = yaml.Unmarshal(ovirtSecretObj.Data[ovirtSecretKey], &ovirtSecretDataMap)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	if _, ok := ovirtSecretDataMap["caCert"]; !ok {
+		return reconcile.Result{}, fmt.Errorf("oVirt secret must contain caCert attribute")
+	}
+	if len(ovirtSecretDataMap["caCert"]) == 0 {
+		return reconcile.Result{}, fmt.Errorf("oVirt secret caCert cannot be empty")
+	}
+	caCert, err := base64.StdEncoding.DecodeString(ovirtSecretDataMap["caCert"])
```

Comment:
I wonder why not to use `data` instead of `stringData`. It would make k8s to decode the content automatically.

Reply:
There are 2 issues here:
1. How to support multiple providers within the same key: data expects the input to be base64 encoded, it will not support providing a key, the same way as stringData does. we'll have to define two sets of variables on the same level prefixed differently for ovirt_ and vmware_ which isn't aligned with the design (differentiate between providers by sectiions)

2. How to represent multi-line value in yaml - seems the convention is to represent it as a wrapped base64 string which is the reason for doing it.

Reply:
what is wrong with using `|`? It seems to be used quite often in Ansible context. Requiring some values encoded and some not seems confusing.